src/sidebars/search/search-controller.js

```diff
@@ -19,15 +19,23 @@
             this._watch();
         }
 
+        hasStore() {
```

Comment:
Looking through I believe the only use of the store is the `get` method, and after the check that is what we do (since saving new favourites is the responsibility of the star controller I think). Maybe that can be incorporated into the new method?